### DISCUSSION TITLE
Will gno communicate with external exchanges with IBC or API call?

### DISCUSSION THREAD

**Discussion by LeeBlues** [gnolang/gno#18] [Q&A] [answered]
Hello Jae and all.  I'm studying and following this project, but I'm sorry that I don't have the skills to contribute to the project. Today I am curious about  platform developers opinion, so I am leaving a post here. I have a contract that I want to develop since Ethereum was released in the past. It can be seen as a decentralized version of the Gamestop incident that took place a while ago. The Gamestop contract works like this:
 Participants deposit money into the contract. if total money exceeded certain amount within certain time frame, then contract purchases a designated coin at a margin from designated exchanges in each time period randomly. If the margin transaction is not liquidated, the purchased coins will be being holt for a certain period of time then it will be distributed to participants.
 //genkidama contract? 
type Gamestop contract{
     coinName  ,  // name of  coin this contract will purchase 
     contractDuration // 
     times ,   // how many times contract will make a buy order
     thresholdMoney // required money to make this contract work
     thresholdTime // thresholdMoney should be collected until this time otherwise, money will be refunded 
     holdperiod// if margin buying was survived, it will be holt
     sendersAddrs[]
     exchanges [] 
 }
  However, I think the difficulty of developing the required platform is incredibly difficult. The question is,
1) Will the gno platform enable IBC communication or API calls with external exchanges?
2) will the exchange accept the smart contract as an independent transaction agent? I think it would be impossible for an existing centralized exchanges that requires KYC. Is this possible with a decentralized exchange developed with Tendermint ? I will ask it to Binance DEX too. 
 Have a nice day all!

**Comment by faddat** [comment]
1) Gno will be able to interface with external exchanges and entities, I am not sure what the exact protocol for doing that will be yet, and I'm not sure if @jaekwon is super-clear on that at this time, either. 

2) Gno realms are (in my understanding) very much like standard computer programs, run in a distributed fashion. 

It is amusing, @jaekwon once asked me about smart contracts, what I thought, and I said that scope restriction was critical. 

....then I see this, which basically gives contracts the full capabilities of your computer with relatively few tradeoffs and little scope restriction.  

Of course, there's a whole lo-fi conception of computing in here, as well. 

Regarding exchanges and KYC, I figure that's up to any of them.  The networks talk but the operators of nodes of course can and do define who they talk to and what they can say to each other. 

Personally, it's my hope that networks of connected chains will make exchanges much less relevant to cryptocurrency / decentralized computing.

But of course that won't happen overnight.

**Reply by LeeBlues** [reply]
thanks! BTW, are you faddat from bitshares? (EDIT) after read code , I concluded that this contract only can be implemented on gno thanks!

**Reply by faddat** [reply]
:D

I am faddat from Bitshares -- that is a blast from the past!

How'd you know of me from the Bitshares context?

That was a long time ago.

**Reply by LeeBlues** [reply]
bitShares holder backthen and still now haha Its been long time!

**Comment by moul** [comment, marked as answer]
Gno will support IBC or an updated version of IBC in the future.

There is no plan to let Gno make any other external API call right now.
